buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -226,46 +287,81 @@ synchronized void start() {
             "install", "--batch", plugin.toString())
         );
 
+        if (keystoreSettings.isEmpty() == false) {
+            checkSuppliers("Keystore", keystoreSettings);
+            runElaticsearchBinScript("elasticsearch-keystore", "create");
+            keystoreSettings.forEach((key, value) -> {
+                runElaticsearchBinScriptWithInput(value.get().toString(), "elasticsearch-keystore", "add", "-x", key);
+            });
+        }
+
         startElasticsearchProcess();
     }
 
+    private void runElaticsearchBinScriptWithInput(String input, String tool, String... args) {
+        try (InputStream byteArrayInputStream = new ByteArrayInputStream(input.getBytes(StandardCharsets.UTF_8))) {
+            services.loggedExec(spec -> {
+                spec.setEnvironment(getESEnvironment());
+                spec.workingDir(workingDir);
+                spec.executable(
+                    OS.conditionalString()
+                        .onUnix(() -> "./bin/" + tool)
```

Comment:
We should probably add some methods to `Conditional` that simply accept `T` instead of `Supplier<T>` as it's unnecessary to make this evaluation lazy in almost every case given that calling `supply()` is going to evaluate those functions immediately.

Reply:
the lazy evaluation is indeed not necessary, but not sure it's worth the code duplication  it brings.